src/app/service/jobs.service.ts

```diff
@@ -336,3 +310,6 @@ export class JobsService {
 
     }
 }
+function showWarning(toast: ToastService, job: JobItem<any>, error){
```

Comment:
I'd suggest moving this function to the abstract JobItem class as a protected function and then calling this.showWarning from the Implementations.